src/main/java/fr/insee/genesis/infrastructure/utils/FileUtils.java

```diff
@@ -152,4 +158,58 @@ public String getSpecFolder(String campaign) {
 	public String getDoneFolder(String campaign, String dataSource) {
 		return  String.format("%s/%s/%s/%s", dataFolderSource, "DONE", dataSource, campaign);
 	}
+
+	/**
+	 * Write a text file.
+	 * @param filePath Path to the file.
+	 * @param fileContent Content of the text file.
+	 */
+	public void writeFile(Path filePath, String fileContent) {
+		Path path = Path.of(dataFolderSource, filePath.toString());
+		boolean fileCreated = false;
+		File myFile = null;
+		try {
+			Files.createDirectories(path.getParent());
+			myFile = path.toFile();
+			fileCreated = myFile.createNewFile();
+		} catch (IOException e) {
+			log.error("Permission refused to create folder: " + path.getParent(), e);
+		}
+		if (!fileCreated){return ;}
+		try (FileWriter myWriter = new FileWriter(myFile)) {
+			myWriter.write(fileContent);
+			log.info(String.format("Text file: %s successfully written", filePath));
+		} catch (IOException e) {
+			log.warn(String.format("Error occurred when trying to write file: %s", filePath), e);
+		}
+	}
+
+	/**
+	 * Appends a JSON object array into file.
+	 * Creates the files if it doesn't exist
+	 * @param filePath Path to the file.
+	 * @param jsonArray JSON objects to write
+	 */
+	public void appendJSONFile(Path filePath, JSONArray jsonArray) {
+		String content = jsonArray.toJSONString();
+        File myFile;
+		try {
+			Files.createDirectories(filePath.getParent());
+			myFile = filePath.toFile();
+
+			try (RandomAccessFile raf = new RandomAccessFile(myFile, "rw")) {
```

Comment:
try in try...

Reply:
I chose to leave the code as it is, as the refactors added unnecessary complexity in my opinion